api/spec/v1.yaml

```diff
@@ -1341,12 +1341,12 @@ components:
           description: Version of the `context`.
         address_data:
           type: string
-          format: hex
+          format: byte
           description: |
-            The hex-encoded data from which the oasis address was derived.
+            The base64-encoded data from which the oasis address was derived.
```

Comment:
Yeah. `format: hex` is not an official openapi format. openapi allows you to put any value for `format`, but unless it's one of the few official ones, it's just informative.
`format: byte` is one of the official ones; it means "the contents are a raw byte sequence, and clients should encode/decode it with base64.

This goes against what would be natural for eth addresses. If this field always contained an eth address, I would say screw it and do the hex serialization by hand. But this field can also contain ed25519 pubkeys (and maybe runtime hashes in some cases? not sure). So the natural lowest common denominator is raw bytes.
In my understanding, if the autogenerated TS client is openapi compliant, it will actually parse the base64 automatically and give consumers the raw bytes, which is not too bad even for eth addresses.

Reply:
got it, thanks!